src/probnum/randprocs/markov/integrator/_integrator.py

```diff
@@ -4,67 +4,106 @@
+    def select_derivative(self, state, derivative):
 
-        Computes the matrix
+        # Once we allow changed orderings, extend this functionality here.
+        # Due to the behaviour in __init__,
+        # non-coordinate-representation should be impossible
+        assert self.state_ordering == "coordinate"
 
-        .. math:: H_i = \\left[ I_d \\otimes e_i \\right] P^{-1},
-
-        where :math:`e_i` is the :math:`i` th unit vector,
-        that projects to the :math:`i` th coordinate of a vector.
-        If the ODE is multidimensional, it projects to **each** of the
-        :math:`i` th coordinates of each ODE dimension.
+        derivative_indices = np.arange(
+            start=derivative, stop=self.state_dimension, step=(self.num_derivatives + 1)
+        )
+        return np.take(state, indices=derivative_indices)
 
-        Parameters
-        ----------
-        coord : int
-            Coordinate index :math:`i` which to project to.
-            Expected to be in range :math:`0 \\leq i \\leq q + 1`.
+    def derivative_selection_operator(self, derivative):
 
-        Returns
-        -------
-        np.ndarray, shape=(d, d*(q+1))
-            Projection matrix :math:`H_i`.
-        """
-        projvec1d = np.eye(self.num_derivatives + 1)[:, coord]
-        projmat1d = projvec1d.reshape((1, self.num_derivatives + 1))
-        projmat = np.kron(np.eye(self.wiener_process_dimension), projmat1d)
-        return projmat
+        selection_unit_vector = np.eye(self.num_derivatives + 1)[:, derivative]
+        selection_unit_vector_as_matrix = selection_unit_vector.reshape(
+            (1, self.num_derivatives + 1)
+        )
+        selection_matrix = np.kron(
+            np.eye(self.wiener_process_dimension), selection_unit_vector_as_matrix
+        )
+        return selection_matrix
+
+    def reorder_state(self, state, current_ordering, target_ordering):
+        """Change e.g. coordinate-wise ordering to derivative-wise ordering."""
+
+        # Check for admissibility of inputs
+        supported_orders = ["derivative", "coordinate"]
+        current_ok = current_ordering in supported_orders
+        target_ok = target_ordering in supported_orders
+        if not current_ok or not target_ok:
+            msg1 = "Reordering is not supported for given keys"
+            msg2 = f" '{current_ordering}' and '{target_ordering}'. "
+            msg3 = "Only combinations of 'derivative' and 'coordinate' are supported."
+            raise ValueError(msg1 + msg2 + msg3)
+
+        # Early exit if no reordering is to be done
+        if current_ordering == target_ordering:
+            return state
+
+        if current_ordering == "coordinate":
+            assert target_ordering == "derivative"
+            return self._reorder_state_from_coordinate_to_derivative(state=state)
+
+        assert current_ordering == "derivative"
+        assert target_ordering == "coordinate"
```

Comment:
a little paranoid, but it might not hurt especially since this also serves as documentation for what happens in this block?